### PR TITLE
rswift 5.2.2

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
==> /usr/local/Cellar/rswift/5.2.1/bin/rswift --version
Error: rswift: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: </5\.2\.1/> was expected to be =~
<"Unknown\n">.
```

**Comment by Bo98** [comment]
I know we often say version tests are rubbish, this one actually does indicate a (minor) problem. The version is added in a GitHub actions release process, so does not exist in a Git clone.

Try use the tarball at https://github.com/mac-cain13/R.swift/releases instead of a tag & revision.

**Comment by chenrui333** [comment]
> I know we often say version tests are rubbish, this one actually does indicate a (minor) problem. The version is now added in a GitHub actions release process, so does not exist in a Git clone - and will not in the future.
> 
> Try use the zip archive at https://github.com/mac-cain13/R.swift/releases instead of a tag & revision.

Yeah, the version check test cannot imply the functional issue with the artifact (that is the push back I saw the most), it does surface the build issue in the past and this one (the version did not get applied during the build process).

Another thing I see is the version check should not be the blocker of the merge process, but we should certainly improve the artifact (which causes less bad UX).

**Comment by chenrui333** [comment]
In this formula, I did not see the reason that we need to stick with the tag-revision, let's see how the build goes with the tarball format.

**Comment by Bo98** [comment]
Oh I misread - the `zip` is a binary archive not a source one. But yeah, downloading a tarball is usually quicker than a git clone.

Seems like they do this:

https://github.com/mac-cain13/R.swift/blob/a2ed5ea368eecb7df6530abee74c6943358c77d8/.github/workflows/release.yml#L19

which is a little messy

**Comment by tomasharkema** [comment]
Please don't use a tarball since we are moving towards a notarized binary built by GitHub Actions. We will provide a proper signed pkg with every release, so maybe that's something to look in to. 

Please see https://github.com/mac-cain13/R.swift/pull/644

**Comment by Bo98** [comment]
We only build things from source in homebrew-core.

**Comment by tomasharkema** [comment]
Thanks for your answer. We'll take this into consideration.

**Comment by tomasharkema** [comment]
FWIW please keep an eye out to this pull request. I've added an extra step to extract a tarball which contains the right version in source. I'll take it up with the maintainer to release this with version 5.2.2.

https://github.com/mac-cain13/R.swift/pull/643

**Comment by tomasharkema** [comment]
We will after merging https://github.com/mac-cain13/R.swift/pull/643 provide a separate tarball with the updated source. Please see: release.tar.gz. Hope this will help.

https://github.com/mac-cain13/R.swift/releases/tag/v5.2.2-alpha.10 
https://github.com/mac-cain13/R.swift/releases/download/v5.2.2-alpha.10/release.tar.gz

**Comment by chenrui333** [comment]
@tomasharkema since you guys are doing the automated update flow, I wonder if you can create a new release to test the flow. 

I am more than happy to close this PR in favor of that.

File: Formula/rswift.rb [hunk 1]
```diff
@@ -1,14 +1,12 @@
 class Rswift < Formula
   desc "Get strong typed, autocompleted resources like images, fonts and segues"
   homepage "https://github.com/mac-cain13/R.swift"
-  url "https://github.com/mac-cain13/R.swift.git",
-      :tag      => "v5.2.0",
-      :revision => "ec748b51d016e42016c93f8df8acb5e8f3635ccc"
+  url "https://github.com/mac-cain13/R.swift/archive/v5.2.1.tar.gz"
```
Comment (tomasharkema) [inline Formula/rswift.rb:4]: When 5.2.2 is out, we could update this to `url "https://github.com/mac-cain13/R.swift/releases/download/v5.2.2/release.tar.gz"`

**Comment by tomasharkema** [comment]
Thanks! We are in the process of updating the flow. I think we will do this fairly quickly so we can use this PR to figure things out and maybe eventually make this the push to 5.2.2.

**Comment by chenrui333** [comment]
works for me!

**Comment by tomasharkema** [comment]
Allright, https://github.com/mac-cain13/R.swift/releases/tag/v5.2.2 is released. Could you update the PR?

File: Formula/rswift.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/rswift.rb]: Thanks!

**Comment by chenrui333** [comment]
@tomasharkema still run into the version unknown issue.

**Comment by chenrui333** [comment]
Just found that I was using a wrong one and updated.

**Comment by chenrui333** [comment]
looks like pretty good, 🚀 

Thanks @@tomasharkema!

**Pull request closed without merging**